### PR TITLE
chore: Upgrade replicache to v12b

### PR DESCRIPTION
**Pull request by arv**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,17 +17,17 @@
     "next": "^12.1.6",
     "react": "17.0.2",
     "react-dom": "17.0.2",
-    "replicache": "^11.0.0",
-    "replicache-nextjs": "^0.4.0",
-    "replicache-react": "^2.9.0",
+    "replicache": "file:../../replicache-internal/replicache-12.0.0.tgz",
```
Comment (aboodman) [inline package.json]: Were you trying to test in preview? If so you'll have to add this file to the branch.

File: src/todo.ts [hunk 2]
```diff
@@ -5,10 +5,10 @@
 import { ReadTransaction } from "replicache";
 
 export type Todo = {
-  id: string;
-  text: string;
-  completed: boolean;
-  sort: number;
+  readonly id: string;
```
Comment (aboodman) [inline src/todo.ts:8]: Is this required for compat with v12 or just cleanup?

File: tsconfig.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
     "allowJs": true,
     "skipLibCheck": true,
     "strict": true,
+    "exactOptionalPropertyTypes": true,
```
Comment (aboodman) [inline tsconfig.json:8]: We should definitely put this in the release notes.

File: src/todo.ts (hunk 2 above)
Comment (arv) [reply src/todo.ts:8]: Cleanup to make things more consistent

File: package.json (hunk 1 above)
Comment (arv) [reply package.json]: No. I tested locally. Once pushed to npm I will update all these PRs

**Pull request merged**